src/ucp/proto/proto_init.h

```diff
@@ -44,32 +45,42 @@ void ucp_proto_common_init_base_caps(
 
 
 /*
- * Accepts list of operations that happen in parallel and create a ranges
- * on min..max which represents their overall performance.
+ * Accepts a list of performance functions for a given range and appends the
+ * convex or concave envelope of these functions to an output list.
+ *
+ * @param [in] perf_list       List of performance functions.
+ * @param [in] range_start     Range interval start
+ * @param [in] range_end       Range interval end.
+ * @param [in] convex          Whether to select convex (maximal) or concave
+ *                             (minimal) function from 'perf_list'.
```

Comment:
maybe replace "function" by "trend", because we couldn't select "convex"/"concave" from linear functions (which are both convex and concave), but used points of the linear functions could give non-linear function.
wdyt?

Reply:
actually, we do select a function: for each range interval, the result is "index" of the function from "perf_list" that is min/max in this interval.